hieroplatform/src/main/java/org/hiero/dataset/RemoteDataSet.java

```diff
@@ -18,87 +18,208 @@
+import org.hiero.pb.PartialResponse;
+import org.hiero.remoting.*;
 import rx.Observable;
-import scala.concurrent.Await;
-import scala.concurrent.Future;
-import scala.concurrent.duration.Duration;
+import rx.subjects.PublishSubject;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.hiero.remoting.HieroServer.DEFAULT_IDS_INDEX;
 
 /**
- * An IDataSet that is a proxy for a DataSet on a remote machine.
+ * An IDataSet that is a proxy for a DataSet on a remote machine. The remote IDataSet
+ * is pointed to by (serverEndpoint, remoteHandle). Any RemoteDataSet instantiated
+ * with a wrong value for either entry of the tuple will result in an exception.
  */
 public class RemoteDataSet<T> implements IDataSet<T> {
-    protected final static int TIMEOUT_MS = 1000;  // TODO: import via config file
-    protected static final Duration duration = Duration.create(TIMEOUT_MS, "milliseconds");
-    protected final ActorRef clientActor;
-    protected final ActorRef remoteActor;
-
-    public RemoteDataSet(
-            final ActorRef clientActor, final ActorRef remoteActor) {
-        this.clientActor = clientActor;
-        this.remoteActor = remoteActor;
+    protected final static int TIMEOUT = 1000;  // TODO: import via config file
+    protected final int remoteHandle;
+    protected final HostAndPort serverEndpoint;
+    private final HieroServerGrpc.HieroServerStub stub;
+
+    public RemoteDataSet(final HostAndPort serverEndpoint) {
+        this(serverEndpoint, DEFAULT_IDS_INDEX);
+    }
+
+    public RemoteDataSet(final HostAndPort serverEndpoint, final int remoteHandle) {
+        this.serverEndpoint = serverEndpoint;
+        this.remoteHandle = remoteHandle;
+        this.stub = HieroServerGrpc.newStub(NettyChannelBuilder
+                .forAddress(serverEndpoint.getHost(), serverEndpoint.getPort())
+                .usePlaintext(true)
+                .build());
     }
 
+    /**
+     * Map operations on a RemoteDataSet result in only one onNext
+     * invocation that will return the final IDataSet.
+     */
     @Override
     public <S> Observable<PartialResult<IDataSet<S>>> map(final IMap<T, S> mapper) {
         final MapOperation<T, S> mapOp = new MapOperation<T, S>(mapper);
-        final Future<Object> future = Patterns.ask(this.clientActor, mapOp, TIMEOUT_MS);
-        try {
-            @SuppressWarnings("unchecked")
-            final Observable<PartialResult<IDataSet<S>>> obs =
-                (Observable<PartialResult<IDataSet<S>>>) Await.result(future, duration);
-            return obs;
-        } catch (final Exception e) {
-            return Observable.error(e);
-        }
+        final byte[] serializedOp = SerializationUtils.serialize(mapOp);
+        final Command command = Command.newBuilder()
+                                       .setIdsIndex(this.remoteHandle)
+                                       .setSerializedOp(ByteString.copyFrom(serializedOp))
```

Comment:
This is converting a byte[] to a ByteString that is required for the protobuf format.